src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java

```diff
@@ -459,6 +461,23 @@ private void processIter(DetailAST root, AstState astState) {
         }
     }
 
+    /**
+     * Creates a new {@link TreeSet} with a deterministic order based on the
+     * Check's name.
+     *
+     * @return The new {@link TreeSet}.
+     */
+    private static Set<AbstractCheck> createNewTreeSetCheck() {
+        return new TreeSet<>((AbstractCheck check1, AbstractCheck check2) -> {
+            int result = check1.getClass().getName().compareTo(check2.getClass().getName());
+            if (result == 0) {
+                result = Integer.compare(check1.hashCode(), check2.hashCode());
```

Comment:
This is required because a set is unique and just enforcing by check name means duplicate check instances are dropped. See https://github.com/checkstyle/checkstyle/issues/9251#issuecomment-774703173 .

The only other way to resolve this is to convert the `Set` to a `List` and do sorts of all the lists before the first usage. There is no `SortedList` class, so there is no implementation to do it automatically.